src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -190,7 +191,13 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     catch (Exception ex)
                     {
                         _fetchMetadataFailure = ex;
-                        _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
+                        _bootstrapRefreshRetryCount = (_bootstrapRefreshRetryCount < int.MaxValue) ? _bootstrapRefreshRetryCount + 1 : int.MaxValue;
```

Comment:
Perhaps we should use https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked.increment

I think so long as we set a reasonable upper bound on the max retires this check isn't necessary. Just increment and move on, if someone has incremented to int max here even with only 1 sec of wait time which we'd never want to go below...that's at least 68 years and no one is going to have that kind of uptime (besides we'd all be retired by then :D)

Reply:
Adoped exponential backoff